src/interfaces/pyo3/mod.rs

```diff
@@ -0,0 +1,35 @@
+use pyo3::{pymodule, types::PyModule, wrap_pyfunction, PyResult, Python};
+
+use self::{
+    generate_gpsw_keys::{generate_master_keys, generate_user_private_key, rotate_attributes},
+    hybrid_gpsw_aes::{
+        decrypt, decrypt_hybrid_block, decrypt_hybrid_header, encrypt, encrypt_hybrid_block,
+        encrypt_hybrid_header, get_encrypted_header_size,
+    },
+};
+use crate::error::FormatErr;
+
+impl From<FormatErr> for pyo3::PyErr {
+    fn from(e: FormatErr) -> Self {
+        pyo3::exceptions::PyTypeError::new_err(format!("{e}"))
+    }
+}
+
+/// A Python module implemented in Rust.
+#[pymodule]
+fn abe_gpsw(_py: Python, m: &PyModule) -> PyResult<()> {
+    m.add_function(wrap_pyfunction!(generate_master_keys, m)?)?;
+    m.add_function(wrap_pyfunction!(generate_user_private_key, m)?)?;
+    m.add_function(wrap_pyfunction!(rotate_attributes, m)?)?;
+    m.add_function(wrap_pyfunction!(get_encrypted_header_size, m)?)?;
+    m.add_function(wrap_pyfunction!(encrypt_hybrid_header, m)?)?;
+    m.add_function(wrap_pyfunction!(decrypt_hybrid_header, m)?)?;
+    m.add_function(wrap_pyfunction!(encrypt_hybrid_block, m)?)?;
+    m.add_function(wrap_pyfunction!(decrypt_hybrid_block, m)?)?;
```

Comment:
I am not entirely sure we should expose these functions to the user. Is it worth it? Don't `encrypt` and `decrypt` suffice?

Anyway, if you want to keep them exposed publicly, I think we should use private indirection for performance, as exposed Py interfaces cannot take slices. Today, by calling these functions, we are doing copies, that could be avoid by using non-`#[pyfunction]`, when calling them internally.
Example [here](https://github.com/Cosmian/abe_gpsw/pull/30/files#diff-c77a1d0fa135bf37d0ea3e5fff00e169bd76fdad2ccf852bb19a551087037b8dR247), where `header` is copied to be passed to `decrypt_hybrid_header`, or [here](https://github.com/Cosmian/abe_gpsw/pull/30/files#diff-c77a1d0fa135bf37d0ea3e5fff00e169bd76fdad2ccf852bb19a551087037b8dR252) where `ciphertext` is also copied.